blueocean-events/src/main/java/io/jenkins/blueocean/events/BlueMessageEnricher.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package io.jenkins.blueocean.events;
+
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.model.ItemGroup;
+import io.jenkins.blueocean.rest.hal.Link;
+import io.jenkins.blueocean.service.embedded.rest.OrganizationImpl;
+import io.jenkins.blueocean.service.embedded.rest.PipelineImpl;
+import jenkins.model.ParameterizedJobMixIn;
+import org.jenkins.pubsub.EventProps;
+import org.jenkins.pubsub.Events;
+import org.jenkins.pubsub.JobChannelMessage;
+import org.jenkins.pubsub.Message;
+import org.jenkins.pubsub.MessageEnricher;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject;
+
+import javax.annotation.Nonnull;
+
+/**
+ * @author <a href="mailto:tom.fennelly@gmail.com">tom.fennelly@gmail.com</a>
+ */
+@Extension
+public class BlueMessageEnricher extends MessageEnricher {
+    
+    enum BlueEventProps {
+        blueocean_job_rest_url,
+        blueocean_job_pipeline_name,
+        blueocean_job_branch_name,
+    }
+    
+    @Override
+    public void enrich(@Nonnull Message message) {
```

Comment:
How is BlueMessageEnricher used? That is where does it get called from?

There are assumptions in the code that might not hold in all cases. 

I think it is probably going to work better if we have BO specific MessageEnricher, this way we can work with it cleanly:

```
BlueMessageEnricher extends MessageEnricher {
    public void enrich(@Nonnull Message message, BluePipeline pipeline);
}

// With pipeline, you get things like pipeline.getOrganization(), will work in all cases, like multi organization, pipeline instance of 
// BlueBranch or BlueFolderPipeline or BlueMultiMasterPipeline etc.
// There is code in there that are directly accessing Jenkins Job object, so better to put them in 
// impl to allow other kind of implementations. If we want to keep it default impl, we should try to see if BluePipeline can be used 
// instead of Jenkins Job
MessageEnricherImpl extends BlueMessageEnricher{

    public void enrich(@Nonnull Message message, BluePipeline pipeline){
        ...
    }

}

```